src/main/java/com/shaft/api/RestActions.java

```diff
@@ -523,116 +253,376 @@ public static String getResponseXMLValue(Object response, String xmlPath) {
+     *
+     * @param response              the full response object returned by
+     *                              performRequest method.
+     * @param referenceJsonFilePath the full absolute path to the test data file
+     *                              that will be used as a reference for this
+     *                              comparison
+     * @param comparisonType        ComparisonType.EQUALS, CONTAINS, MATCHES,
+     *                              EQUALS_STRICT; Note that MATCHES ignores the
+     *                              content ordering inside the JSON
+     * @return a boolean value that is TRUE in case the comparison passed, or FALSE
+     * in case it failed
+     */
+    public static boolean compareJSON(Response response, String referenceJsonFilePath, ComparisonType comparisonType) {
+        return compareJSON(response, referenceJsonFilePath, comparisonType, "");
+    }
+
+    /**
+     * Compares the Response object against the content of the referenceJsonFilePath
+     *
+     * @param response              the full response object returned by
+     *                              performRequest method.
+     * @param referenceJsonFilePath the full absolute path to the test data file
+     *                              that will be used as a reference for this
+     *                              comparison
+     * @param comparisonType        ComparisonType.EQUALS, CONTAINS; Note that
+     *                              MATCHES ignores the content ordering inside the
+     *                              JSON
+     * @param jsonPathToTargetArray a jsonpath that will be parsed to point to the
+     *                              target JSON Array
+     * @return a boolean value that is TRUE in case the comparison passed, or FALSE
+     * in case it failed
+     */
+    public static boolean compareJSON(Response response, String referenceJsonFilePath, ComparisonType comparisonType,
+                                      String jsonPathToTargetArray) {
+        if (jsonPathToTargetArray.equals("")) {
+            ReportManager.logDiscrete("Comparing the provided API response with the file at this path \""
+                    + referenceJsonFilePath + "\", comparison type \"" + comparisonType + "\"");
+        } else {
+            ReportManager.logDiscrete("Comparing the provided API response with the file at this path \""
+                    + referenceJsonFilePath + "\", comparison type \"" + comparisonType
+                    + "\", jsonPath to target array \"" + jsonPathToTargetArray + "\".");
+        }
+        boolean comparisonResult;
+        JSONParser parser = new JSONParser();
+        List<Object> expectedJSONAttachment = null;
+
+        try {
+            // parse actual JSON into Object or Array
+            org.json.simple.JSONObject actualJsonObject = null;
+            org.json.simple.JSONArray actualJsonArray = null;
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'org.json.simple.JSONArray' due to existing import 'org.json.simple.JSONArray'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5737302)